components/http_client.rst

```diff
@@ -0,0 +1,457 @@
+(if the number of redirects is higher than the configured value, you'll get a
+:class:`Symfony\\Component\\HttpClient\\Exception\\RedirectionException`)::
+
+    $response = $httpClient->request('GET', 'https://...', [
+        // 0 means to not follow any redirect
+        'max_redirects' => 0,
+    ]);
+
+Concurrent Requests
+~~~~~~~~~~~~~~~~~~~
+
+
+.. TODO
+
+
+Asynchronous Requests
+~~~~~~~~~~~~~~~~~~~~~
+
+
+.. TODO  see https://gist.github.com/tgalopin/a84a11ece0621b8a79ed923afe015b3c
+
+
+Processing Responses
+--------------------
+
+The response returned by all HTTP clients is an object of type
+:class:`Symfony\\Contracts\\HttpClient\\ResponseInterface` which provides the
+following methods::
+
+    $response = $httpClient->request('GET', 'https://...');
+
+    // gets the HTTP status code of the response
+    $statusCode = $response->getStatusCode();
+
+    // gets the HTTP headers as string[][] with the header names lower-cased
+    $headers = $response->getHeaders();
+
+    // gets the response body as a string
+    $content = $response->getContent();
+
+    // returns info coming from the transport layer, such as "raw_headers",
+    // "redirect_count", "start_time", "redirect_url", etc.
+    $httpInfo = $response->getInfo();
+    // you can get individual info too
+    $startTime = $response->getInfo('start_time');
+
+Streaming Responses
+~~~~~~~~~~~~~~~~~~~
+
+
+.. TODO:
+
+
+Handling Exceptions
+~~~~~~~~~~~~~~~~~~~
+
+When an HTTP error happens (status code 3xx, 4xx or 5xx) your code is expected
+to handle it by checking the status code of the response. If you don't do that,
+the ``getHeaders()`` and ``getContent()`` methods throw an appropriate exception::
+
+    $response = $httpClient->request('GET', 'https://httpbin.org/status/403');
+
+    // this code results in a Symfony\Component\HttpClient\Exception\ClientException
+    // because it doesn't check the status code of the response
+    $content = $response->getContent();
+
+    // pass FALSE as the optional argument to not throw an exception and
+    // return instead an empty string
+    $content = $response->getContent(false);
+
+PSR-7 and PSR-18 Compatibility
+------------------------------
+
+This component uses its own interfaces and exception classes different from the
+ones defined in `PSR-7`_ (HTTP message interfaces) and `PSR-18`_ (HTTP Client).
+However, it includes the :class:`Symfony\\Component\\HttpClient\\Psr18Client`
+class, which is an adapter to turn a Symfony ``HttpClientInterface`` into a
+PSR-18 ``ClientInterface``.
+
+Before using it in your app, run the following commands to install the required
+dependencies:
+
+.. code-block:: terminal
+
+    # installs the base ClientInterface
+    $ composer require psr/http-client
+
+    # installs an efficient implementation of response and stream factories
+    # with autowiring aliases provided by Symfony Flex
+    $ composer require nyholm/psr7
+
+Symfony Framework Integration
```

Comment:
I think this section shouldn't be part of the standalone component page.

Reply:
According to what we did in the past, you are right. However, this is one of those components where "component only = 99% of its docs" and "component + Symfony = 1% of its docs" ... so creating a short article about "HttpClient in a Symfony app" would be confusing and counterproductive.

If the integration of the component in Symfony keeps being a minimal section, I'd say we should keep it in this article.